internal/server/kong/ws/config/compat/compatibility_table.go

```diff
@@ -869,6 +869,36 @@ var (
 				RemoveFields: []string{"http_response_header_for_traceid"},
 			},
 		},
+		{
+			Metadata: config.ChangeMetadata{
+				ID:          config.ChangeID("P140"),
+				Severity:    config.ChangeSeverityError,
+				Description: standardCoreEntityMessage("key", "3.1"),
+				Resolution:  standardUpgradeMessage("3.1"),
+			},
+			SemverRange: versionsPre310,
+			Update: config.ConfigTableUpdates{
+				Name:         "keys",
+				Type:         config.TopLevel,
+				RemoveFields: []string{"keys"},
+				Remove:       true,
```

Comment:
We should test this extensively, but at a first glance this is mixing 2 concept we have been using in 2 different ways: `RemoveFields` is used to remove some entity fields. while `Remove` is used to remove the whole entity. I guess that the latter is useless after your introduction of `TopLevel`?

Reply:
i couldn't find a way to remove a whole "thing", so i added that `TopLevel`, which is a different code, so it does reuse other fields for a slightly different purpose.  in this case, those are the fields of the top level to remove.  the `Remove` field is unused right now.

(would it be too bad taste to say that all this makes me miss XSLT?)

Reply:
Could we not [utilize the current implementation that exists](https://github.com/Kong/koko/blob/f761e0bd4ba0437f40c5e6504c62065adc496f75/internal/server/kong/ws/config/version_compatibility.go#L32-L52) and define `Keys` and `KeySets` instead of `TopLevel`?

Reply:
didn't realize the `Remove` flag actually removed the whole type of entities.

done